core/src/main/java/com/graphhopper/routing/lm/PrepareLandmarks.java

```diff
@@ -135,40 +125,11 @@ public void doSpecificWork() {
                 + ", nodes:" + graph.getNodes() + ", " + Helper.getMemInfo());
     }
 
-    public RoutingAlgorithm getPreparedRoutingAlgorithm(Graph qGraph, RoutingAlgorithm algo, AlgorithmOptions opts) {
-        int activeLM = Math.max(1, opts.getHints().getInt(Landmark.ACTIVE_COUNT, defaultActiveLandmarks));
-        if (algo instanceof AStar) {
-            if (!lms.isInitialized())
-                throw new IllegalStateException("Initialize landmark storage before creating algorithms");
-
-            double epsilon = opts.getHints().getDouble(Parameters.Algorithms.AStar.EPSILON, 1);
-            AStar astar = (AStar) algo;
-            astar.setApproximation(new LMApproximator(qGraph, lmProfile.getWeighting(), this.graph.getNodes(), lms, activeLM, lms.getFactor(), false).
-                    setEpsilon(epsilon));
-            return algo;
-        } else if (algo instanceof AStarBidirection) {
-            if (!lms.isInitialized())
-                throw new IllegalStateException("Initialize landmark storage before creating algorithms");
-
-            double epsilon = opts.getHints().getDouble(Parameters.Algorithms.AStarBi.EPSILON, 1);
-            AStarBidirection astarbi = (AStarBidirection) algo;
-            astarbi.setApproximation(new LMApproximator(qGraph, lmProfile.getWeighting(), this.graph.getNodes(), lms, activeLM, lms.getFactor(), false).
-                    setEpsilon(epsilon));
-            return algo;
-        } else if (algo instanceof AlternativeRoute) {
-            if (!lms.isInitialized())
-                throw new IllegalStateException("Initialize landmark storage before creating algorithms");
-
-            double epsilon = opts.getHints().getDouble(Parameters.Algorithms.AStarBi.EPSILON, 1);
-            AlternativeRoute altRoute = (AlternativeRoute) algo;
-            altRoute.setApproximation(new LMApproximator(qGraph, lmProfile.getWeighting(), this.graph.getNodes(), lms, activeLM, lms.getFactor(), false).
-                    setEpsilon(epsilon));
-            // landmark algorithm follows good compromise between fast response and exploring 'interesting' paths so we
-            // can decrease this exploration factor further (1->dijkstra, 0.8->bidir. A*)
-            altRoute.setMaxExplorationFactor(0.6);
-        }
-
-        return algo;
+    /**
+     * Convenience method to obtain a routing algo factory from the preparation.
+     */
+    public RoutingAlgorithmFactory getRoutingAlgorithmFactory(int defaultActiveLandmarks) {
```

Comment:
Having the defaultActiveLandmarks parameter here is a bit ugly. Maybe we can derive some sane default from landmark storage instead (and provide an optional setter on the routing factory or something?)

Reply:
Yeah, the defaultActiveLandmarks should probably guessed from the landmark storage (or be half of all landmarks but at least 12 or something)

Reply:
Ok I think as long as we keep this parameter it should be a property of the LM routing algorithm factory, but I removed it from the constructor parameter list and set it to a default as you suggested. It can be changed using a setter.